packages/core/src/rules/common/component-name-unique.ts

```diff
@@ -0,0 +1,158 @@
+import { Problem, UserContext } from '../../walk';
+import { Oas2Rule, Oas3Rule, Oas3Visitor } from '../../visitors';
+import {
+  Oas3Definition,
+  Oas3Parameter,
+  Oas3RequestBody,
+  Oas3Response,
+  Oas3Schema,
+  OasRef,
+} from '../../typings/openapi';
+import { Oas2Definition } from '../../typings/swagger';
+
+const TYPE_NAME_SCHEMA = 'Schema';
```

Comment:
Why do you need this mapping? For me it looks like it only adds additional complexity 🙂

Reply:
I wonder if [mapTypeToComponent](https://github.com/Redocly/redocly-cli/blob/0ce9cab59a72fec7ae43e103a79207e4fd6eef67/packages/core/src/bundle.ts#L191) could help with the mapping 🤔